### PR TITLE
Added possiblity to define healthCheck to custom path

### PR DESCRIPTION
**Pull request by lukidzi**
Some applications expose different port/path for healthChecks. Currently it isn't possible to define  custom `healthCheck` to `local_service` for service. This PR introduce possibility to define custom healthCheck - path and clusterName definiton and send it in metadata. You should define cluster in your envoy configuration. In future it will be reasonable to send more information like healthCheck port to EC and generate cluster rather then writing static configuration in file.

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt [hunk 1]
```diff
@@ -182,6 +201,13 @@ data class TimeoutPolicy(
     val responseTimeout: Duration?
 )
 
+data class HealthCheck(
+    val path: String = "",
+    val clusterName: String = "local_service"
+) {
+    fun hasHealthCheck() = !path.isBlank()
```
Comment (slonka) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt]: I don't get it, the default HC does not have a health check? (explaination: path="", path.isBlank() == true, hasHealthCheck = false).

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataTest.kt [hunk 2]
```diff
@@ -181,4 +181,71 @@ class NodeMetadataTest {
...
+        assertThat(incoming.healthCheck.path).isEqualTo("/status/ping")
+        assertThat(incoming.healthCheck.hasHealthCheck()).isTrue()
+    }
+
+    @Test
+    fun `should set empty healthCheck path for incoming settings when path is empty`() {
+        // given
+        val proto = proxySettingsProto(
+            incomingSettings = true,
+            path = "/path"
```
Comment (slonka) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataTest.kt]: Path is not empty here, healthCheckPath is. Title a little bit misleading / or variable name.

File: envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml [hunk 3]
```diff
@@ -0,0 +1,127 @@
+admin:
```
Comment (slonka) [inline envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml]: I think we need to trim down the configs size in general.

File: envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      incoming:
+        healthCheck:
+          path: "/status/custom"
+          clusterName: "local_service_health_check"
+        endpoints:
+          - path: "/endpoint"
+            clients: ["authorizedClient"]
+      outgoing:
+        dependencies:
+          - service: "service-1"
```
Comment (slonka) [inline envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml]: I do not think you need all of the dependencies in your test.

File: envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  metadata:
+    ads: true
+    proxy_settings:
+      incoming:
+        healthCheck:
+          path: "/status/custom"
+          clusterName: "local_service_health_check"
+        endpoints:
+          - path: "/endpoint"
+            clients: ["authorizedClient"]
```
Comment (slonka) [inline envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml]: You don't need authorized client.

File: envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+          stat_prefix: egress_http
+          rds:
+            route_config_name: default_routes
+            config_source:
+              ads: {}
+          http_filters:
+          - name: envoy.filters.http.header_to_metadata
+            config:
+              request_rules:
+                - header: x-canary
```
Comment (slonka) [inline envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml]: Canary also not needed I think.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyAdmin.kt [hunk 7]
```diff
@@ -44,7 +44,11 @@ class EnvoyAdmin(
             }
 
     fun statValue(statName: String): String? = get("stats?filter=$statName").body()?.use {
-        it.string().lines().first().split(":").get(1).trim()
+        val splitedStats = it.string().lines().first().split(":")
+        if (splitedStats.size != 2) {
+            return "0"
```
Comment (slonka) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyAdmin.kt]: Are we sure? Some tests rely on 0 value, maybe -1?

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt (hunk 1 above)
Comment (lukidzi) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt]: It is custom healthCheck. By default EC sets healthCheck to path defined in application properties and `local_service` cluster.

Comment (lukidzi) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt]: I'll change the name to `hasCustomHealthCheck`.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyAdmin.kt (hunk 7 above)
Comment (lukidzi) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyAdmin.kt]: Done

File: envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml (hunk 6 above)
Comment (lukidzi) [reply envoy-control-tests/src/main/resources/envoy/config_ads_custom_health_check.yaml]: Simplified config

**Pull request merged**